default/methods/source_control/file/file_browser.py

```diff
@@ -664,59 +704,33 @@ def file_view_core(
-            working_dir_file_list, count = self.build_and_execute_query(
-                limit = self.metadata["limit"],
-                offset = self.metadata["start_index"],
-            )
-            if regular_log.log_has_error(self.log):
-                return None
-            if not working_dir_file_list or regular_log.log_has_error(self.log):
-                return False
-            file_count += count
-        else:
-            policy_engine = PolicyEngine(session = self.session, project = self.project)
-            perm_result = policy_engine.member_has_perm(
-                member = self.member,
-                object_id = self.directory.id,
-                object_type = ValidObjectTypes.dataset,
-                perm = DatasetPermissions.dataset_view
-            )
-            if not perm_result.allowed:
-                self.log['error']['unauthorized'] = f'Cannot view dataset ID: {self.directory.id}'
-                resp = jsonify(log=self.log)
-                resp.status = 401
-                raise Unauthorized(response = resp)
-
-            query, count = WorkingDirFileLink.file_list(
-                session = self.session,
-                working_dir_id = self.directory.id,
-                ann_is_complete = ann_is_complete,
-                type = media_type_query,
-                return_mode = "query",
-                limit = self.metadata["limit"],
-                date_from = self.metadata["date_from"],
-                date_to = self.metadata["date_to"],
-                issues_filter = self.metadata["issues_filter"],
-                offset = self.metadata["start_index"],
-                original_filename = self.metadata['search_term'],
-                order_by_class_and_attribute = File.id,
-                order_by_direction = order_by_direction,
-                exclude_removed = exclude_removed,
-                file_view_mode = self.metadata['file_view_mode'],
-                job_id = job_id,
-                has_some_machine_made_instances = has_some_machine_made_instances,
-                ignore_id_list = ignore_id_list,
-                count_before_limit = True,
-                include_children_compound = self.metadata['with_children_files']
-
-            )
-
-            file_count += count
-
-            working_dir_file_list = query.all()
+        query, count = WorkingDirFileLink.file_list(
+            session = self.session,
+            working_dir_id = self.directory.id,
+            ann_is_complete = ann_is_complete,
+            type = media_type_query,
+            return_mode = "query",
+            limit = self.metadata["limit"],
+            date_from = self.metadata["date_from"],
+            date_to = self.metadata["date_to"],
+            issues_filter = self.metadata["issues_filter"],
+            offset = self.metadata["start_index"],
+            original_filename = self.metadata['search_term'],
+            order_by_class_and_attribute = File.id,
```

Comment:
I noticed we have logic to determine `order_by_class_and_attribute` above, but then we dont use the result and just set the field to `File.id`. Is this a bug or on purpose?